test/terra/backends/aer_simulator/test_wrapper_statevector_simulator.py

```diff
@@ -23,252 +24,264 @@
-from qiskit.providers.aer import StatevectorSimulator
+from qiskit import transpile
+from qiskit.providers.aer import StatevectorSimulator, AerError
+from test.terra.backends.simulator_test_case import (
+    SimulatorTestCase, supported_devices)
 
 
-class StatevectorSimulatorTests:
+@ddt
+class TestStatevectorSimulator(SimulatorTestCase):
     """StatevectorSimulator tests."""
 
-    SIMULATOR = StatevectorSimulator()
-    BACKEND_OPTS = {}
+    BACKEND = StatevectorSimulator
 
     # ---------------------------------------------------------------------
     # Test initialize
     # ---------------------------------------------------------------------
-    def test_initialize_1(self):
+
+    @supported_devices
+    def test_initialize_1(self, device):
         """Test StatevectorSimulator initialize"""
+        backend = self.backend(device=device)
         circuits = ref_initialize.initialize_circuits_1(final_measure=False)
         targets = ref_initialize.initialize_statevector_1()
-        qobj = assemble(circuits, shots=1)
-        sim_job = self.SIMULATOR.run(qobj, **self.BACKEND_OPTS)
-        result = sim_job.result()
+        result = backend.run(circuits).result()
         self.assertSuccess(result)
         self.compare_statevector(result, circuits, targets)
 
-    def test_initialize_2(self):
+    @supported_devices
+    def test_initialize_2(self, device):
         """Test StatevectorSimulator initialize"""
+        backend = self.backend(device=device)
         circuits = ref_initialize.initialize_circuits_2(final_measure=False)
         targets = ref_initialize.initialize_statevector_2()
-        qobj = assemble(circuits, shots=1)
-        sim_job = self.SIMULATOR.run(qobj, **self.BACKEND_OPTS)
-        result = sim_job.result()
+        result = backend.run(circuits).result()
         self.assertSuccess(result)
         self.compare_statevector(result, circuits, targets)
 
     # ---------------------------------------------------------------------
     # Test reset
     # ---------------------------------------------------------------------
-    def test_reset_deterministic(self):
+    @supported_devices
+    def test_reset_deterministic(self, device):
         """Test StatevectorSimulator reset with for circuits with deterministic counts"""
         # For statevector output we can combine deterministic and non-deterministic
         # count output circuits
+        backend = self.backend(device=device)
         circuits = ref_reset.reset_circuits_deterministic(final_measure=False)
-        targets = ref_reset.reset_statevector_deterministic()
-        job = execute(circuits,
-                      self.SIMULATOR,
-                      shots=1,
-                      **self.BACKEND_OPTS)
-        result = job.result()
+        circuits = transpile(circuits, backend, optimization_level=1)
+        result = backend.run(circuits, shots=1).result()
+        targets = ref_reset.reset_statevector_deterministic()        
```

Comment:
There's a bunch of extra whitespace on this line, and on the similar line in the next test case.